sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -637,3 +652,314 @@ def test_categories(density, drop):
 @pytest.mark.parametrize('Encoder', [OneHotEncoder, OrdinalEncoder])
 def test_encoders_has_categorical_tags(Encoder):
     assert 'categorical' in Encoder()._get_tags()['X_types']
+
+
+@pytest.mark.parametrize("kwargs", [
+    {'max_levels': 2},
+    {'min_frequency': 11},
+    {'min_frequency': 0.29},
+    {'max_levels': 2, 'min_frequency': 6},
+    {'max_levels': 4, 'min_frequency': 12},
+])
+@pytest.mark.parametrize("categories",
+                         ["auto", [['a', 'b', 'c', 'd']]])
+def test_ohe_infrequent_two_levels(kwargs, categories):
+
+    X_train = np.array([['a'] * 5 + ['b'] * 20 + ['c'] * 10 + ['d'] * 3]).T
+    ohe = OneHotEncoder(categories=categories,
+                        handle_unknown='auto', sparse=False,
+                        **kwargs).fit(X_train)
+    assert_array_equal(ohe.infrequent_indices_, [[0, 2, 3]])
+
+    X_test = [['b'], ['a'], ['c'], ['d'], ['e']]
+    expected = np.array([
+        [1, 0],
+        [0, 1],
+        [0, 1],
+        [0, 1],
+        [0, 1]])
+
+    X_trans = ohe.transform(X_test)
+    assert_allclose(expected, X_trans)
+
+    expected_inv = [['b'], ['c'], ['c'], ['c'], ['c']]
```

Comment:
why are the infrequent cats  transformed back to c, and not e.g. a or d? is this documented?

Reply:
This is now documented in the `inverse_transform` method. The `get_feature_names` is documented in the user guide and the method.